.github/workflows/node.yml

```diff
@@ -177,4 +177,82 @@ jobs:
 
             - name: Test compatibility
               run: npm test -- -t "set and get flow works"
-              working-directory: ./node
+              working-directory: ./node                   
+
+    start-self-hosted-runner:
+        if: github.repository_owner == 'aws'
+        runs-on: ubuntu-latest
+        steps:
+          - name: Checkout
+            uses: actions/checkout@v4
+
+          - name: Start self hosted EC2 runner
+            uses: ./.github/workflows/start-self-hosted-runner
+            with:
+                aws-access-key-id: ${{ secrets.AWS_EC2_ACCESS_KEY_ID }}
+                aws-secret-access-key: ${{ secrets.AWS_EC2_SECRET_ACCESS_KEY }}
+                aws-region: ${{ secrets.AWS_REGION }}
+                ec2-instance-id: ${{ secrets.AWS_EC2_INSTANCE_ID }}
+
+    checkout-self-hosted-runner:
+      if: github.repository_owner == 'aws'
+      runs-on: [self-hosted, Linux, ARM64]
+      needs: start-self-hosted-runner
+      steps:
+        - name: Setup self-hosted runner access
+          run: sudo chown -R $USER:$USER /home/ubuntu/actions-runner/_work/glide-for-redis
+        - name: Checkout
+          uses: actions/checkout@v4
+      
+    build-and-test-linux-musl-latest:
```

Comment:
why checkout-self-hosted-runner and build-and-test-linux-musl-latest are in separate jobs? 
why not having a single test-linux-musl-latest job that do all -checkout, build and test - same as we do for other platforms (e.g. for ubuntu)?

Reply:
oh, I see that it's bc build-and-test-linux-musl-latest runs inside the container.
Can you add a comment explaining what's the purpose of checkout-self-hosted-runner into the code?